astropy/units/equivalencies.py

```diff
@@ -558,6 +559,68 @@ def beam_angular_area(beam_area):
             (astrophys.Jy/astrophys.beam, astrophys.Jy/Unit(beam_area)),
            ]
 
+
+def thermodynamic_temperature(frequency, T_cmb=None):
+    r"""Defines the conversion between Jy/beam and "thermodynamic temperature",
+    :math:`T_{CMB}`, in Kelvins.  The thermodynamic temperature is a unit very
+    commonly used in cosmology. See eqn 8 in [1]
+
+    :math:`K_{CMB} \equiv I_\nu / \left(2 k \nu^2 / c^2  f(\nu) \right)`
+
+    with :math:`f(\nu) = \frac{ x^2 e^x}{(e^x - 1 )^2}`
+    where :math:`x = h \nu / k T`
+
+    Parameters
+    ----------
+    frequency : `~astropy.units.Quantity` with spectral units
+        The observed `spectral` equivalent `~astropy.units.Unit` (e.g.,
+        frequency or wavelength)
+    T_cmb :  `~astropy.units.Quantity` with temperature units (default Planck15 value)
+        The CMB temperature at z=0
+
+    Notes
+    -----
+    For broad band receivers, this conversion do not hold
+    as it highly depends on the frequency
+
+    References
+    ----------
+    .. [1] Planck 2013 results. IX. HFI spectral response
+       https://arxiv.org/abs/1303.5070
+
+    Examples
+    --------
+    Planck HFI 143 GHz::
+
+        >>> import numpy as np
+        >>> from astropy import units as u
+        >>> freq = 143*u.GHz
+        >>> equiv = u.thermodynamic_temperature(freq)
+        >>> u.K.to(u.MJy/u.sr, equivalencies=equiv)  # doctest: +FLOAT_CMP
```

Comment:
Nitpicky detail, but maybe convert an actual quantity in the example:
```
(1.*u.mK).to(u.MJy/u.sr, equivalencies=equiv)
```
(I *think* that is a more logical number to convert...)